src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs

```diff
@@ -2215,7 +2229,7 @@ private unsafe void ReplaceInPlaceAtChunk(ref StringBuilder chunk, ref int index
                     indexInChunk += lengthToCopy;
                     if (indexInChunk >= chunk.m_ChunkLength)
                     {
-                        chunk = Next(chunk);
+                        chunk = Next(chunk)!;
```

Comment:
Why is this one ok to suppress?  It made sense in the previous function, because the very next line checked for null and bailed... what's makes it ok here?

Reply:
I think in here this is different because we already checked all the ranges and made room before calling this so we should never get past the last chunk. I'll create an intermediate variable and add assert